aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -0,0 +1,653 @@
+#error "Big endian is not supported."
+#endif
+
+template<int index, bool mask_val>
+struct BlendRegs {
+  static float32x4_t impl(
+    const float32x4_t& a, const float32x4_t& b, float32x4_t& res);
+};
+
+template<int index>
+struct BlendRegs<index, true>{
+  static float32x4_t impl(
+      const float32x4_t& a, const float32x4_t& b, float32x4_t& res) {
+    return vsetq_lane_f32(vgetq_lane_f32(b, index), res, index);
+  }
+};
+
+template<int index>
+struct BlendRegs<index, false>{
+  static float32x4_t impl(
+      const float32x4_t& a, const float32x4_t& b, float32x4_t& res) {
+    return vsetq_lane_f32(vgetq_lane_f32(a, index), res, index);
+  }
+};
+
+template <> class Vec256<float> {
+private:
+  float32x4x2_t values;
+public:
+  using value_type = float;
+  static constexpr int size() {
+    return 8;
+  }
+  Vec256() {}
+  Vec256(float32x4x2_t v) : values(v) {}
+  Vec256(float val) {
+    values.val[0] = vdupq_n_f32(val);
+    values.val[1] = vdupq_n_f32(val);
+  }
+  Vec256(float val0, float val1, float val2, float val3,
+         float val4, float val5, float val6, float val7) {
+    values.val[0] = {val0, val1, val2, val3};
+    values.val[1] = {val4, val5, val6, val7};
+  }
+  Vec256(float32x4_t val0, float32x4_t val1) {
+    values.val[0] = val0;
+    values.val[1] = val1;
+  }
+  operator float32x4x2_t() const {
+    return values;
+  }
+  template <int64_t mask>
+  static Vec256<float> blend(const Vec256<float>& a, const Vec256<float>& b) {
+    Vec256<float> vec;
+    // 0.
+    vec.values.val[0] =
+      BlendRegs<0, (mask & 0x01)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    vec.values.val[0] =
+      BlendRegs<1, (mask & 0x02)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    vec.values.val[0] =
+      BlendRegs<2, (mask & 0x04)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    vec.values.val[0] =
+      BlendRegs<3, (mask & 0x08)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    // 1.
+    vec.values.val[1] =
+      BlendRegs<0, (mask & 0x10)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    vec.values.val[1] =
+      BlendRegs<1, (mask & 0x20)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    vec.values.val[1] =
+      BlendRegs<2, (mask & 0x40)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    vec.values.val[1] =
+      BlendRegs<3, (mask & 0x80)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    return vec;
+  }
+  static Vec256<float> blendv(const Vec256<float>& a, const Vec256<float>& b,
+                              const Vec256<float>& mask) {
+    // TODO
+    // NB: This requires that each value, i.e., each uint value,
+    // of the mask either all be zeros or all be 1s.
+    // We perhaps need some kind of an assert?
+    // But that will affect performance.
```

Comment:
Asserts should be compiled out in release builds.  Having a check in debug builds is worthwhile I would say.

Reply:
Do you mean TORCH_INTERNAL_ASSERT?